buildsystem/HandlePythonOptions.cmake

```
@@ -4,7 +4,17 @@
 
 # the Python version number requirement is in modules/FindPython_test.cpp
 find_package(Python ${PYTHON_MIN_VERSION} REQUIRED)
-find_package(Cython ${CYTHON_MIN_VERSION} REQUIRED)
+find_package(Cython 3.0.10 QUIET)
+if (CYTHON_FOUND AND CYTHON_VERSION VERSION_GREATER_EQUAL 3.0.10)
+    message("Using Cython >= 3.0.10 (found version ${CYTHON_VERSION})")
+else()
```

```suggestion
else()
    message("Cython version ${CYTHON_MIN_VERSION} not available. Searching for ${CYTHON_MIN_VERSION_FALLBACK} instead")
```

You can add a failback message before the next `find_package` call here.

---

Added